### DISCUSSION TITLE
`strictToken` doesn't play nicely when dealing with multiple animation names

### DISCUSSION THREAD

**Discussion by Newbie012** [chakra-ui/chakra-ui#9122] [General]
### Description

When I tried to `animationName="fade-in, scale-in"` while using `strictTokens`, I expected the string to be valid, but instead I got the following TSC error:

```
Type '"fade-in, scale-in"' is not assignable to type 'ConditionalValue<WithEscapeHatch<"position" | `var(--${string})` | "spin" | "pulse" | "ping" | "bounce" | "bg-position" | "circular-progress" | "expand-height" | "collapse-height" | "expand-width" | "collapse-width" | ... 19 more ... | "scale-out">> | undefined'
```

### Link to Reproduction

https://stackblitz.com/edit/vitejs-vite-qm4gfj?file=src%2FApp.tsx

### Steps to reproduce

1. Run `npm run reproduce`
2. Go to [`App.tsx`](https://stackblitz.com/edit/vitejs-vite-qm4gfj?file=src%2FApp.tsx)
3. See error

### Chakra UI Version

3.0.1

### Browser

Google Chrom 130

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by Newbie012** [comment]
I believe it would require a complex type to support this kind of string literal.

As an alternative, I initially thought about doing `animationName={["fade-in", "scale-in"]}` but it collides with the [responsive array syntax](https://next.chakra-ui.com/docs/styling/responsive-design#array-syntax).

Another alternative would be to support a new syntax, like `animationName={{ multiple: ["fade-in", "scale-in"] }}` (or maybe a better name).

My last alternative would be to add a custom ESLint rule that enforces complex cases based on the config file.

**Comment by segunadebayo** [comment]
Indeed, this is a complex scenario that requires special handling.

I'll move this to a discussion for now so the community can weigh-in on the best approach. I'll give it some thought as well.

For now, I'd say the ESLint approach is probably good.

**Comment by segunadebayo** [comment]
Closing as stale. Please start a new discussion if this is still relevant on the latest version.
